src/main/java/seedu/address/model/person/ClassTiming.java

```diff
@@ -45,10 +48,87 @@ public String formatClassTiming(String classTiming) {
     }
 
     /**
-     * Returns true if a given string is a valid phone number.
+     * Converts the day into the int representation of the day.
+     *
+     * @param day String value of the day in class timing.
+     * @return int representatin of the days of the week.
+     */
+    public int replaceDayWithInt(String day) {
+        switch (day) {
+        case "MON":
+            return 1;
+        case "TUE":
+            return 2;
+        case "WED":
+            return 3;
+        case "THU":
+            return 4;
+        case "FRI":
+            return 5;
+        case "SAT":
+            return 6;
+        case "SUN":
+            return 7;
+        default:
+            return 0;
+        }
+    }
+
+    /**
+     * Compares two ClassTiming objects.
+     *
+     * @param otherClassTiming ClassTiming being compared to.
+     * @return true if this ClassTiming is on an earlier day or has end time earlier than otherClassTiming
+     * start time, otherwise false.
+     */
+    public boolean isEarlier(ClassTiming otherClassTiming) {
+        String[] thisClassTimingPart = this.value.split(" ");
+        String thisDay = thisClassTimingPart[0];
+        int thisDayInt = replaceDayWithInt(thisDay);
+        LocalTime thisStartTime = getStartTime(this.value);
+        LocalTime thisEndTime = getEndTime(this.value);
+
+        String[] otherClassTimingPart = otherClassTiming.value.split(" ");
+        String otherDay = otherClassTimingPart[0];
+        int otherDayInt = replaceDayWithInt(otherDay);
+        LocalTime otherStartTime = getStartTime(otherClassTiming.value);
+
+
+        if (thisDayInt < otherDayInt || otherStartTime.isAfter(thisEndTime)
+                || thisStartTime.isBefore(otherStartTime)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static LocalTime getStartTime(String ct) {
+        String[] ctPart = ct.split(" ");
+        String startEndTime = ctPart[1];
+        String[] timePart = startEndTime.split("-");
```

Comment:
This part is repeated on line 114 to 116, maybe it would be neater if it could be refactored into another method.

Reply:
Other than this, LGTM!

Reply:
hmm im not sure, since the method is to already get start/end time, if I refactor that part out it would be like gettiming, sounds abit weird to me, would it be too many layers